avroregistry/registry_test.go

```diff
@@ -179,6 +181,34 @@ func TestCanceledRetry(t *testing.T) {
 	}
 }
 
+func TestUnavailableError(t *testing.T) {
+	c := qt.New(t)
+	defer c.Done()
+	// When the service in unavailable, the response is probably not
+	// formatted as JSON.
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		w.Header().Set("Content-Type", "text/html; charset=UTF-8")
+		w.WriteHeader(500)
+		w.Write([]byte(`
+<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
+<html><head>
+<title>502 Proxy Error</title>
+</head><body>
+<h1>Proxy Error</h1>
+<p>The whole world is bogus
```

Comment:
like the missing `</p>`? :smile:

Reply:
This is HTML not XML - there's no need for a closing tag AFAIK.